# aws cred error when aws is not used

**tomdavidson** commented *Nov 11, 2017*

Im loving how responsible the issues and I am hope I'm not being an annoying newb!

I am working on using tg in ci/cd  to deliver k8s clusters to my team. I have scrapped what I was doing pre tg and have reorganized the structure to be tg compatible.  I have super simple mock iac plans that Im using to pave the workflow. Looking very promising. Unfortunately:

```sh
$ terragrunt validate-all --terragrunt-working-dir stage
[terragrunt] [stage] 2017/11/10 18:05:26 Running command: terraform --version
[terragrunt] 2017/11/10 18:05:51 Error processing module at 'devops-k8s/stage/aws/terraform.tfvars'. How this module was found: Terragrunt config file found in a subdirectory of stage. Underlying error: NoCredentialProviders: no valid providers in chain. Deprecated.
        For verbose messaging see aws.Config.CredentialsChainVerboseErrors
[terragrunt] 2017/11/10 18:05:51 Unable to determine underlying exit code, so Terragrunt will exit with error code 1
```

The odd thing is I have nothing AWS related so why are creds needed?


```
├── stage
│   ├── aws
│   └── vmw
└── tf_modules
    ├── dirname
```
The two stage envs consist of a single terraform.tfvars file with:

```hcl
terragrunt = {
  include {
    path = "${find_in_parent_folders()}"
  }

  terraform {
    source = "../../tf_modules/dirname"
  }
}
```

The dirname module consists of:
```hcl
resource "random_pet" "mod" {}

output "test" {
  value = "${path.module}-${random_pet.mod.id}"
}
```

And my root terraform.tfvars file consists of:

```hcl
terragrunt = {

}
```


<br />
***


**brikis98** commented *Nov 11, 2017*

Weird. Could you set the `TERRAGRUNT_DEBUG` env var so we can see a stack trace?

```bash
export TERRAGRUNT_DEBUG=true
terragrunt validate-all
```
***

**tomdavidson** commented *Nov 11, 2017*

$ terragrunt validate-all --terragrunt-working-dir stage
[terragrunt] [stage] 2017/11/10 19:10:22 Running command: terraform --version
[terragrunt] 2017/11/10 19:10:46 configstack.ErrorProcessingModule Error processing module at 'devops-k8s/stage/aws/terraform.tfvars'. How this module was found: Terragrunt config file found in a subdirectory of stage. Underlying error: NoCredentialProviders: no valid providers in chain. Deprecated.
        For verbose messaging see aws.Config.CredentialsChainVerboseErrors
/home/ubuntu/.go_workspace/src/github.com/gruntwork-io/terragrunt/configstack/module.go:88 (0x4c16cf)
/home/ubuntu/.go_workspace/src/github.com/gruntwork-io/terragrunt/configstack/module.go:64 (0x4c1431)
/home/ubuntu/.go_workspace/src/github.com/gruntwork-io/terragrunt/configstack/module.go:44 (0x4c1181)
/home/ubuntu/.go_workspace/src/github.com/gruntwork-io/terragrunt/configstack/stack.go:127 (0x4c695a)
/home/ubuntu/.go_workspace/src/github.com/gruntwork-io/terragrunt/configstack/stack.go:110 (0x4c65e8)
/home/ubuntu/.go_workspace/src/github.com/gruntwork-io/terragrunt/cli/cli_app.go:546 (0x45e90f)
/home/ubuntu/.go_workspace/src/github.com/gruntwork-io/terragrunt/cli/cli_app.go:447 (0x45de54)
/home/ubuntu/.go_workspace/src/github.com/gruntwork-io/terragrunt/cli/cli_app.go:194 (0x45c584)
/home/ubuntu/.go_workspace/src/github.com/gruntwork-io/terragrunt/cli/cli_app.go:177 (0x45c29c)
/home/ubuntu/.go_workspace/src/github.com/gruntwork-io/terragrunt/vendor/github.com/urfave/cli/app.go:502 (0x49c5e4)
/home/ubuntu/.go_workspace/src/github.com/gruntwork-io/terragrunt/vendor/github.com/urfave/cli/app.go:268 (0x49a31d)
/home/ubuntu/.go_workspace/src/github.com/gruntwork-io/terragrunt/main.go:20 (0x4010c4)
/usr/local/go/src/runtime/proc.go:183 (0x42a864)
/usr/local/go/src/runtime/asm_amd64.s:2086 (0x458e51)

[terragrunt] 2017/11/10 19:10:46 Unable to determine underlying exit code, so Terragrunt will exit with error code 1
***

**tomdavidson** commented *Nov 11, 2017*

OK, Iv been caught. My root terrafrom.tfvar looked like:

```hcl
terragrunt = {

  # remote_state {
  #   backend = "s3"
  #   config {
  #     bucket         = "devops-k8s-${get_env("TF_VAR_ENV", "exp")}-${get_env("TF_VAR_REGION", "us-west-2")}-${get_aws_account_id()}"
  #     key            = "${path_relative_to_include()}/terraform.tfstate"
  #     region         = "${get_env("TF_VAR_REGION", "us-west-2")}"
  #     encrypt        = true
  #     dynamodb_table = "devops-k8s-${get_env("TF_VAR_ENV", "exp")}-${get_env("TF_VAR_REGION", "us-west-2")}-${get_aws_account_id()}"
  #   }
 # }
}
```

After removing the commented code. I do not get the aws cred error.  Comments are not supported?
***

**brikis98** commented *Nov 11, 2017*

Ah, that's a bug. While Terragrunt uses the HCL parser (which does ignore comments) to parse the `.tfvars` file, it first uses simple regex to process "interpolations" (e.g., `${...}`) within the HCL. My guess is that regex is picking up the `get_aws_account_id()` call, even though it's in comments. As discussed in #339, we need to change the parse order so it does HCL first to get the structure, and only then processes the interpolations within that structure.
***

